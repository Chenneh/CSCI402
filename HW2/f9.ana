
==== Packets ====
p1: inter-arrival time should be 413.076ms
    time in Q1 should be 419.754ms
    time in Q2 should be 0.029ms
    should request 1429ms of service time
    time in S2 should be 1431.606ms
    time in system should be 1851.485ms

p2: inter-arrival time should be 419.890ms
    time in Q1 should be 831.459ms
    time in Q2 should be 0.029ms
    should request 1429ms of service time
    time in S1 should be 1441.391ms
    time in system should be 2272.911ms

p3: inter-arrival time should be 412.832ms
    time in Q1 should be 1251.786ms
    time in Q2 should be 0.010ms
    should request 1429ms of service time
    time in S2 should be 1448.221ms
    time in system should be 2700.026ms

p4: inter-arrival time should be 418.713ms
    time in Q1 should be 1665.609ms
    time in Q2 should be 0.016ms
    should request 1429ms of service time
    time in S1 should be 1446.439ms
    time in system should be 3112.090ms

p5: inter-arrival time should be 416.906ms
    time in Q1 should be 2069.951ms
    time in Q2 should be 0.012ms
    should request 1429ms of service time
    time in S2 should be 1464.613ms
    time in system should be 3534.584ms

p6: inter-arrival time should be 416.150ms
    time in Q1 should be 2475.889ms
    time in Q2 should be 0.012ms
    should request 1429ms of service time
    time in S1 should be 1445.393ms
    time in system should be 3921.307ms

p7: inter-arrival time should be 416.275ms
    time in Q1 should be 2899.885ms
    time in Q2 should be 0.012ms
    should request 1429ms of service time
    time in S2 should be 1436.304ms
    time in system should be 4336.209ms

p8: inter-arrival time should be 416.332ms
    time in Q1 should be 3298.515ms
    time in Q2 should be 0.024ms
    should request 1429ms of service time
    time in S1 should be 1436.937ms
    time in system should be 4735.486ms

p9: inter-arrival time should be 416.756ms
    time in Q1 should be 3713.748ms
    time in Q2 should be 0.013ms
    should request 1429ms of service time
    time in S2 should be 1445.779ms
    time in system should be 5159.549ms

p10: inter-arrival time should be 404.421ms
    time in Q1 should be 4133.416ms
    time in Q2 should be 0.016ms
    should request 1429ms of service time
    time in S1 should be 1446.194ms
    time in system should be 5579.645ms

p11: inter-arrival time should be 405.260ms
    time in Q1 should be 4543.914ms
    time in Q2 should be 0.013ms
    should request 1429ms of service time
    time in S2 should be 1439.824ms
    time in system should be 5983.766ms

p12: inter-arrival time should be 416.832ms
    time in Q1 should be 4948.931ms
    time in Q2 should be 0.013ms
    should request 1429ms of service time
    time in S1 should be 1445.910ms
    time in system should be 6394.869ms

p13: inter-arrival time should be 418.775ms
    time in Q1 should be 5349.775ms
    time in Q2 should be 0.013ms
    should request 1429ms of service time
    time in S2 should be 1445.259ms
    time in system should be 6795.079ms

p14: inter-arrival time should be 421.491ms
    time in Q1 should be 5763.547ms
    time in Q2 should be 0.014ms
    should request 1429ms of service time
    time in S1 should be 1436.509ms
    time in system should be 7200.085ms

p15: inter-arrival time should be 405.156ms
    time in Q1 should be 6182.227ms
    time in Q2 should be 0.013ms
    should request 1429ms of service time
    time in S2 should be 1444.577ms
    time in system should be 7626.841ms

==== Statistics ====
average packet inter-arrival time = 0.414591
average packet service time = 1.44366373333333
average number of packets in Q1 = 3.57858434016004
average number of packets in Q2 = 1.7261537279287e-05
average number of packets in S1 = 0.729373835207354
average number of packets in S2 = 0.834633822848382
average time a packet spent in system = 4.7469288
standard deviation for time spent in system = 1.77582008351358
token drop probability = 0
packet drop probability = 0
