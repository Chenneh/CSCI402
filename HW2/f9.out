Emulation Parameters:
	number to arrive = 15
	lambda = 2.5
	mu = 0.7
	r = 2.5
	B = 7
	P = 2
00000000.000ms: emulation begins
00000413.001ms: token t1 arrives, token bucket now has 1 token 
00000413.076ms: p1 arrives, needs 2 tokens, inter-arrival time = 413.076ms
00000413.079ms: p1 enters Q1
00000832.822ms: token t2 arrives, token bucket now has 2 tokens
00000832.833ms: p1 leaves Q1, time in Q1 = 419.752ms, token bucket now has 0 token 
00000832.859ms: p1 enters Q2
00000832.888ms: p1 leaves Q2, time in Q2 = 0.029ms
00000832.955ms: p1 begin service at S2, requesting 1428ms of service
00000832.966ms: p2 arrives, needs 2 tokens, inter-arrival time = 419.890ms
00000832.969ms: p2 enters Q1
00001245.782ms: token t3 arrives, token bucket now has 1 token 
00001245.798ms: p3 arrives, needs 2 tokens, inter-arrival time = 412.832ms
00001245.801ms: p3 enters Q1
00001664.420ms: token t4 arrives, token bucket now has 2 tokens
00001664.428ms: p2 leaves Q1, time in Q1 = 831.457ms, token bucket now has 0 token 
00001664.432ms: p2 enters Q2
00001664.461ms: p2 leaves Q2, time in Q2 = 0.029ms
00001664.486ms: p2 begin service at S1, requesting 1428ms of service
00001664.511ms: p4 arrives, needs 2 tokens, inter-arrival time = 418.713ms
00001664.514ms: p4 enters Q1
00002081.402ms: token t5 arrives, token bucket now has 1 token 
00002081.417ms: p5 arrives, needs 2 tokens, inter-arrival time = 416.906ms
00002081.420ms: p5 enters Q1
00002264.561ms: p1 departs from S2, servie time = 1431.606ms, time in system = 1851.485ms
00002497.567ms: p6 arrives, needs 2 tokens, inter-arrival time = 416.150ms
00002497.575ms: p6 enters Q1
00002497.585ms: token t6 arrives, token bucket now has 2 tokens
00002497.587ms: p3 leaves Q1, time in Q1 = 1251.784ms, token bucket now has 0 token 
00002497.590ms: p3 enters Q2
00002497.600ms: p3 leaves Q2, time in Q2 = 0.010ms
00002497.603ms: p3 begin service at S2, requesting 1428ms of service
00002913.827ms: token t7 arrives, token bucket now has 1 token 
00002913.842ms: p7 arrives, needs 2 tokens, inter-arrival time = 416.275ms
00002913.845ms: p7 enters Q1
00003105.877ms: p2 departs from S1, servie time = 1441.391ms, time in system = 2272.911ms
00003330.095ms: token t8 arrives, token bucket now has 2 tokens
00003330.123ms: p4 leaves Q1, time in Q1 = 1665.608ms, token bucket now has 0 token 
00003330.141ms: p4 enters Q2
00003330.157ms: p4 leaves Q2, time in Q2 = 0.016ms
00003330.162ms: p4 begin service at S1, requesting 1428ms of service
00003330.174ms: p8 arrives, needs 2 tokens, inter-arrival time = 416.332ms
00003330.177ms: p8 enters Q1
00003746.915ms: token t9 arrives, token bucket now has 1 token 
00003746.930ms: p9 arrives, needs 2 tokens, inter-arrival time = 416.756ms
00003746.933ms: p9 enters Q1
00003945.824ms: p3 departs from S2, servie time = 1448.221ms, time in system = 2700.026ms
00004151.351ms: p10 arrives, needs 2 tokens, inter-arrival time = 404.421ms
00004151.360ms: p10 enters Q1
00004151.368ms: token t10 arrives, token bucket now has 2 tokens
00004151.371ms: p5 leaves Q1, time in Q1 = 2069.949ms, token bucket now has 0 token 
00004151.373ms: p5 enters Q2
00004151.385ms: p5 leaves Q2, time in Q2 = 0.012ms
00004151.388ms: p5 begin service at S2, requesting 1428ms of service
00004556.611ms: p11 arrives, needs 2 tokens, inter-arrival time = 405.260ms
00004556.619ms: p11 enters Q1
00004556.629ms: token t11 arrives, token bucket now has 1 token 
00004776.601ms: p4 departs from S1, servie time = 1446.439ms, time in system = 3112.090ms
00004973.443ms: p12 arrives, needs 2 tokens, inter-arrival time = 416.832ms
00004973.451ms: p12 enters Q1
00004973.461ms: token t12 arrives, token bucket now has 2 tokens
00004973.464ms: p6 leaves Q1, time in Q1 = 2475.887ms, token bucket now has 0 token 
00004973.466ms: p6 enters Q2
00004973.478ms: p6 leaves Q2, time in Q2 = 0.012ms
00004973.481ms: p6 begin service at S1, requesting 1428ms of service
00005392.218ms: p13 arrives, needs 2 tokens, inter-arrival time = 418.775ms
00005392.243ms: p13 enters Q1
00005392.255ms: token t13 arrives, token bucket now has 1 token 
00005616.001ms: p5 departs from S2, servie time = 1464.613ms, time in system = 3534.584ms
00005813.709ms: p14 arrives, needs 2 tokens, inter-arrival time = 421.491ms
00005813.717ms: p14 enters Q1
00005813.727ms: token t14 arrives, token bucket now has 2 tokens
00005813.730ms: p7 leaves Q1, time in Q1 = 2899.883ms, token bucket now has 0 token 
00005813.732ms: p7 enters Q2
00005813.744ms: p7 leaves Q2, time in Q2 = 0.012ms
00005813.747ms: p7 begin service at S2, requesting 1428ms of service
00006218.865ms: p15 arrives, needs 2 tokens, inter-arrival time = 405.156ms
00006218.874ms: p15 enters Q1
00006219.094ms: token t15 arrives, token bucket now has 1 token 
00006418.874ms: p6 departs from S1, servie time = 1445.393ms, time in system = 3921.307ms
00006628.683ms: token t16 arrives, token bucket now has 2 tokens
00006628.692ms: p8 leaves Q1, time in Q1 = 3298.513ms, token bucket now has 0 token 
00006628.696ms: p8 enters Q2
00006628.720ms: p8 leaves Q2, time in Q2 = 0.024ms
00006628.723ms: p8 begin service at S1, requesting 1428ms of service
00007047.995ms: token t17 arrives, token bucket now has 1 token 
00007250.051ms: p7 departs from S2, servie time = 1436.304ms, time in system = 4336.209ms
00007460.673ms: token t18 arrives, token bucket now has 2 tokens
00007460.681ms: p9 leaves Q1, time in Q1 = 3713.746ms, token bucket now has 0 token 
00007460.684ms: p9 enters Q2
00007460.697ms: p9 leaves Q2, time in Q2 = 0.013ms
00007460.700ms: p9 begin service at S2, requesting 1428ms of service
00007878.048ms: token t19 arrives, token bucket now has 1 token 
00008065.660ms: p8 departs from S1, servie time = 1436.937ms, time in system = 4735.486ms
00008284.765ms: token t20 arrives, token bucket now has 2 tokens
00008284.776ms: p10 leaves Q1, time in Q1 = 4133.414ms, token bucket now has 0 token 
00008284.781ms: p10 enters Q2
00008284.797ms: p10 leaves Q2, time in Q2 = 0.016ms
00008284.802ms: p10 begin service at S1, requesting 1428ms of service
00008688.880ms: token t21 arrives, token bucket now has 1 token 
00008906.479ms: p9 departs from S2, servie time = 1445.779ms, time in system = 5159.549ms
00009100.524ms: token t22 arrives, token bucket now has 2 tokens
00009100.533ms: p11 leaves Q1, time in Q1 = 4543.912ms, token bucket now has 0 token 
00009100.537ms: p11 enters Q2
00009100.550ms: p11 leaves Q2, time in Q2 = 0.013ms
00009100.553ms: p11 begin service at S2, requesting 1428ms of service
00009505.113ms: token t23 arrives, token bucket now has 1 token 
00009730.996ms: p10 departs from S1, servie time = 1446.194ms, time in system = 5579.645ms
00009922.373ms: token t24 arrives, token bucket now has 2 tokens
00009922.382ms: p12 leaves Q1, time in Q1 = 4948.929ms, token bucket now has 0 token 
00009922.385ms: p12 enters Q2
00009922.398ms: p12 leaves Q2, time in Q2 = 0.013ms
00009922.402ms: p12 begin service at S1, requesting 1428ms of service
00010324.588ms: token t25 arrives, token bucket now has 1 token 
00010540.377ms: p11 departs from S2, servie time = 1439.824ms, time in system = 5983.766ms
00010742.009ms: token t26 arrives, token bucket now has 2 tokens
00010742.018ms: p13 leaves Q1, time in Q1 = 5349.772ms, token bucket now has 0 token 
00010742.021ms: p13 enters Q2
00010742.034ms: p13 leaves Q2, time in Q2 = 0.013ms
00010742.038ms: p13 begin service at S2, requesting 1428ms of service
00011159.833ms: token t27 arrives, token bucket now has 1 token 
00011368.312ms: p12 departs from S1, servie time = 1445.910ms, time in system = 6394.869ms
00011577.256ms: token t28 arrives, token bucket now has 2 tokens
00011577.264ms: p14 leaves Q1, time in Q1 = 5763.545ms, token bucket now has 0 token 
00011577.268ms: p14 enters Q2
00011577.282ms: p14 leaves Q2, time in Q2 = 0.014ms
00011577.285ms: p14 begin service at S1, requesting 1428ms of service
00011995.182ms: token t29 arrives, token bucket now has 1 token 
00012187.297ms: p13 departs from S2, servie time = 1445.259ms, time in system = 6795.079ms
00012401.093ms: token t30 arrives, token bucket now has 2 tokens
00012401.101ms: p15 leaves Q1, time in Q1 = 6182.225ms, token bucket now has 0 token 
00012401.113ms: p15 enters Q2
00012401.126ms: p15 leaves Q2, time in Q2 = 0.013ms
00012401.129ms: p15 begin service at S2, requesting 1428ms of service
00013013.794ms: p14 departs from S1, servie time = 1436.509ms, time in system = 7200.085ms
00013845.706ms: p15 departs from S2, servie time = 1444.577ms, time in system = 7626.841ms
00013845.812ms: emulation ends
	average packet inter-arrival time time = 0.414591
	average packet service time = 1.44366

	average number of packets in Q1 = 3.57858
	average number of packets in Q2 = 1.72615e-05
	average number of packets in S1 = 0.729374
	average number of packets in S2 = 0.834633
	average time a packet spent in system = 4.74693
	standard deviation for time spent in system = 1.77582

	token drop probability = 0
	packet drop probability = 0

