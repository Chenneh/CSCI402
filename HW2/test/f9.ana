
==== Packets ====
p1: inter-arrival time should be 416.113ms
    time in Q1 should be 403.773ms
    time in Q2 should be 0.022ms
    should request 1429ms of service time
    time in S1 should be 1437.966ms
    time in system should be 1841.827ms

p2: inter-arrival time should be 403.881ms
    time in Q1 should be 827.451ms
    time in Q2 should be 0.015ms
    should request 1429ms of service time
    time in S2 should be 1444.216ms
    time in system should be 2271.699ms

p3: inter-arrival time should be 416.116ms
    time in Q1 should be 1242.261ms
    time in Q2 should be 0.011ms
    should request 1429ms of service time
    time in S1 should be 1435.558ms
    time in system should be 2677.863ms

p4: inter-arrival time should be 411.377ms
    time in Q1 should be 1655.844ms
    time in Q2 should be 0.011ms
    should request 1429ms of service time
    time in S2 should be 1444.806ms
    time in system should be 3100.671ms

p5: inter-arrival time should be 414.944ms
    time in Q1 should be 2073.130ms
    time in Q2 should be 0.009ms
    should request 1429ms of service time
    time in S1 should be 1442.969ms
    time in system should be 3516.117ms

p6: inter-arrival time should be 415.950ms
    time in Q1 should be 2490.135ms
    time in Q2 should be 0.011ms
    should request 1429ms of service time
    time in S2 should be 1434.696ms
    time in system should be 3924.869ms

p7: inter-arrival time should be 408.463ms
    time in Q1 should be 2906.411ms
    time in Q2 should be 0.012ms
    should request 1429ms of service time
    time in S1 should be 1443.715ms
    time in system should be 4350.151ms

p8: inter-arrival time should be 416.471ms
    time in Q1 should be 3299.285ms
    time in Q2 should be 0.014ms
    should request 1429ms of service time
    time in S2 should be 1444.711ms
    time in system should be 4744.025ms

p9: inter-arrival time should be 416.455ms
    time in Q1 should be 3710.440ms
    time in Q2 should be 0.013ms
    should request 1429ms of service time
    time in S1 should be 1428.914ms
    time in system should be 5139.377ms

p10: inter-arrival time should be 415.773ms
    time in Q1 should be 4117.817ms
    time in Q2 should be 0.060ms
    should request 1429ms of service time
    time in S2 should be 1445.503ms
    time in system should be 5563.396ms

p11: inter-arrival time should be 416.974ms
    time in Q1 should be 4523.308ms
    time in Q2 should be 0.048ms
    should request 1429ms of service time
    time in S1 should be 1446.235ms
    time in system should be 5969.603ms

p12: inter-arrival time should be 415.961ms
    time in Q1 should be 4940.471ms
    time in Q2 should be 0.013ms
    should request 1429ms of service time
    time in S2 should be 1428.642ms
    time in system should be 6369.142ms

p13: inter-arrival time should be 408.956ms
    time in Q1 should be 5354.317ms
    time in Q2 should be 0.014ms
    should request 1429ms of service time
    time in S1 should be 1444.197ms
    time in system should be 6798.537ms

p14: inter-arrival time should be 415.810ms
    time in Q1 should be 5761.041ms
    time in Q2 should be 0.013ms
    should request 1429ms of service time
    time in S2 should be 1445.420ms
    time in system should be 7206.490ms

p15: inter-arrival time should be 406.988ms
    time in Q1 should be 6183.568ms
    time in Q2 should be 0.013ms
    should request 1429ms of service time
    time in S1 should be 1433.624ms
    time in system should be 7617.223ms

==== Statistics ====
average packet inter-arrival time = 0.4133488
average packet service time = 1.44007813333333
average number of packets in Q1 = 3.58156732160138
average number of packets in Q2 = 2.01913999978578e-05
average number of packets in S1 = 0.833215706969666
average number of packets in S2 = 0.730074272508924
average time a packet spent in system = 4.73939933333333
standard deviation for time spent in system = 1.77668448079985
token drop probability = 0
packet drop probability = 0
