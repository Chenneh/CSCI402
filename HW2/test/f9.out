Emulation Parameters:
	number to arrive = 15
	lambda = 2.5
	mu = 0.7
	r = 2.5
	B = 7
	P = 2
00000000.000ms: emulation begins
00000416.069ms: token t1 arrives, token bucket now has 1 token 
00000416.113ms: p1 arrives, needs 2 tokens, inter-arrival time = 416.113ms
00000416.116ms: p1 enters Q1
00000819.873ms: token t2 arrives, token bucket now has 2 tokens
00000819.889ms: p1 leaves Q1, time in Q1 = 403.773ms, token bucket now has 0 token 
00000819.944ms: p1 enters Q2
00000819.966ms: p1 leaves Q2, time in Q2 = 0.022ms
00000819.974ms: p1 begin service at S1, requesting 1428ms of service
00000819.994ms: p2 arrives, needs 2 tokens, inter-arrival time = 403.881ms
00000819.996ms: p2 enters Q1
00001236.110ms: p3 arrives, needs 2 tokens, inter-arrival time = 416.116ms
00001236.137ms: p3 enters Q1
00001236.161ms: token t3 arrives, token bucket now has 1 token 
00001647.436ms: token t4 arrives, token bucket now has 2 tokens
00001647.447ms: p2 leaves Q1, time in Q1 = 827.451ms, token bucket now has 0 token 
00001647.451ms: p2 enters Q2
00001647.466ms: p2 leaves Q2, time in Q2 = 0.015ms
00001647.477ms: p2 begin service at S2, requesting 1428ms of service
00001647.487ms: p4 arrives, needs 2 tokens, inter-arrival time = 411.377ms
00001647.490ms: p4 enters Q1
00002062.415ms: token t5 arrives, token bucket now has 1 token 
00002062.431ms: p5 arrives, needs 2 tokens, inter-arrival time = 414.944ms
00002062.434ms: p5 enters Q1
00002257.940ms: p1 departs from S1, servie time = 1437.966ms, time in system = 1841.827ms
00002478.381ms: p6 arrives, needs 2 tokens, inter-arrival time = 415.950ms
00002478.389ms: p6 enters Q1
00002478.395ms: token t6 arrives, token bucket now has 2 tokens
00002478.398ms: p3 leaves Q1, time in Q1 = 1242.261ms, token bucket now has 0 token 
00002478.401ms: p3 enters Q2
00002478.412ms: p3 leaves Q2, time in Q2 = 0.011ms
00002478.415ms: p3 begin service at S1, requesting 1428ms of service
00002886.844ms: p7 arrives, needs 2 tokens, inter-arrival time = 408.463ms
00002886.852ms: p7 enters Q1
00002886.860ms: token t7 arrives, token bucket now has 1 token 
00003091.693ms: p2 departs from S2, servie time = 1444.216ms, time in system = 2271.699ms
00003303.315ms: p8 arrives, needs 2 tokens, inter-arrival time = 416.471ms
00003303.324ms: p8 enters Q1
00003303.332ms: token t8 arrives, token bucket now has 2 tokens
00003303.334ms: p4 leaves Q1, time in Q1 = 1655.844ms, token bucket now has 0 token 
00003303.337ms: p4 enters Q2
00003303.348ms: p4 leaves Q2, time in Q2 = 0.011ms
00003303.352ms: p4 begin service at S2, requesting 1428ms of service
00003719.738ms: token t9 arrives, token bucket now has 1 token 
00003719.770ms: p9 arrives, needs 2 tokens, inter-arrival time = 416.455ms
00003719.773ms: p9 enters Q1
00003913.973ms: p3 departs from S1, servie time = 1435.558ms, time in system = 2677.863ms
00004135.543ms: p10 arrives, needs 2 tokens, inter-arrival time = 415.773ms
00004135.552ms: p10 enters Q1
00004135.561ms: token t10 arrives, token bucket now has 2 tokens
00004135.564ms: p5 leaves Q1, time in Q1 = 2073.130ms, token bucket now has 0 token 
00004135.567ms: p5 enters Q2
00004135.576ms: p5 leaves Q2, time in Q2 = 0.009ms
00004135.579ms: p5 begin service at S1, requesting 1428ms of service
00004552.502ms: token t11 arrives, token bucket now has 1 token 
00004552.517ms: p11 arrives, needs 2 tokens, inter-arrival time = 416.974ms
00004552.520ms: p11 enters Q1
00004748.158ms: p4 departs from S2, servie time = 1444.806ms, time in system = 3100.671ms
00004968.478ms: p12 arrives, needs 2 tokens, inter-arrival time = 415.961ms
00004968.487ms: p12 enters Q1
00004968.522ms: token t12 arrives, token bucket now has 2 tokens
00004968.524ms: p6 leaves Q1, time in Q1 = 2490.135ms, token bucket now has 0 token 
00004968.527ms: p6 enters Q2
00004968.538ms: p6 leaves Q2, time in Q2 = 0.011ms
00004968.554ms: p6 begin service at S2, requesting 1428ms of service
00005377.418ms: token t13 arrives, token bucket now has 1 token 
00005377.434ms: p13 arrives, needs 2 tokens, inter-arrival time = 408.956ms
00005377.437ms: p13 enters Q1
00005578.548ms: p5 departs from S1, servie time = 1442.969ms, time in system = 3516.117ms
00005793.244ms: p14 arrives, needs 2 tokens, inter-arrival time = 415.810ms
00005793.253ms: p14 enters Q1
00005793.260ms: token t14 arrives, token bucket now has 2 tokens
00005793.263ms: p7 leaves Q1, time in Q1 = 2906.411ms, token bucket now has 0 token 
00005793.265ms: p7 enters Q2
00005793.277ms: p7 leaves Q2, time in Q2 = 0.012ms
00005793.280ms: p7 begin service at S1, requesting 1428ms of service
00006200.216ms: token t15 arrives, token bucket now has 1 token 
00006200.232ms: p15 arrives, needs 2 tokens, inter-arrival time = 406.988ms
00006200.235ms: p15 enters Q1
00006403.250ms: p6 departs from S2, servie time = 1434.696ms, time in system = 3924.869ms
00006602.600ms: token t16 arrives, token bucket now has 2 tokens
00006602.609ms: p8 leaves Q1, time in Q1 = 3299.285ms, token bucket now has 0 token 
00006602.612ms: p8 enters Q2
00006602.626ms: p8 leaves Q2, time in Q2 = 0.014ms
00006602.629ms: p8 begin service at S2, requesting 1428ms of service
00007013.653ms: token t17 arrives, token bucket now has 1 token 
00007236.995ms: p7 departs from S1, servie time = 1443.715ms, time in system = 4350.151ms
00007430.204ms: token t18 arrives, token bucket now has 2 tokens
00007430.213ms: p9 leaves Q1, time in Q1 = 3710.440ms, token bucket now has 0 token 
00007430.216ms: p9 enters Q2
00007430.229ms: p9 leaves Q2, time in Q2 = 0.013ms
00007430.233ms: p9 begin service at S1, requesting 1428ms of service
00007843.815ms: token t19 arrives, token bucket now has 1 token 
00008047.340ms: p8 departs from S2, servie time = 1444.711ms, time in system = 4744.025ms
00008253.357ms: token t20 arrives, token bucket now has 2 tokens
00008253.369ms: p10 leaves Q1, time in Q1 = 4117.817ms, token bucket now has 0 token 
00008253.373ms: p10 enters Q2
00008253.433ms: p10 leaves Q2, time in Q2 = 0.060ms
00008253.436ms: p10 begin service at S2, requesting 1428ms of service
00008659.819ms: token t21 arrives, token bucket now has 1 token 
00008859.147ms: p9 departs from S1, servie time = 1428.914ms, time in system = 5139.377ms
00009075.819ms: token t22 arrives, token bucket now has 2 tokens
00009075.828ms: p11 leaves Q1, time in Q1 = 4523.308ms, token bucket now has 0 token 
00009075.832ms: p11 enters Q2
00009075.880ms: p11 leaves Q2, time in Q2 = 0.048ms
00009075.885ms: p11 begin service at S1, requesting 1428ms of service
00009492.233ms: token t23 arrives, token bucket now has 1 token 
00009698.939ms: p10 departs from S2, servie time = 1445.503ms, time in system = 5563.396ms
00009908.950ms: token t24 arrives, token bucket now has 2 tokens
00009908.958ms: p12 leaves Q1, time in Q1 = 4940.471ms, token bucket now has 0 token 
00009908.962ms: p12 enters Q2
00009908.975ms: p12 leaves Q2, time in Q2 = 0.013ms
00009908.978ms: p12 begin service at S2, requesting 1428ms of service
00010324.825ms: token t25 arrives, token bucket now has 1 token 
00010522.120ms: p11 departs from S1, servie time = 1446.235ms, time in system = 5969.603ms
00010731.745ms: token t26 arrives, token bucket now has 2 tokens
00010731.754ms: p13 leaves Q1, time in Q1 = 5354.317ms, token bucket now has 0 token 
00010731.757ms: p13 enters Q2
00010731.771ms: p13 leaves Q2, time in Q2 = 0.014ms
00010731.774ms: p13 begin service at S1, requesting 1428ms of service
00011147.996ms: token t27 arrives, token bucket now has 1 token 
00011337.620ms: p12 departs from S2, servie time = 1428.642ms, time in system = 6369.142ms
00011554.286ms: token t28 arrives, token bucket now has 2 tokens
00011554.294ms: p14 leaves Q1, time in Q1 = 5761.041ms, token bucket now has 0 token 
00011554.298ms: p14 enters Q2
00011554.311ms: p14 leaves Q2, time in Q2 = 0.013ms
00011554.314ms: p14 begin service at S2, requesting 1428ms of service
00011967.362ms: token t29 arrives, token bucket now has 1 token 
00012175.971ms: p13 departs from S1, servie time = 1444.197ms, time in system = 6798.537ms
00012383.795ms: token t30 arrives, token bucket now has 2 tokens
00012383.803ms: p15 leaves Q1, time in Q1 = 6183.568ms, token bucket now has 0 token 
00012383.814ms: p15 enters Q2
00012383.827ms: p15 leaves Q2, time in Q2 = 0.013ms
00012383.831ms: p15 begin service at S1, requesting 1428ms of service
00012999.734ms: p14 departs from S2, servie time = 1445.420ms, time in system = 7206.490ms
00013817.455ms: p15 departs from S1, servie time = 1433.624ms, time in system = 7617.223ms
00013817.764ms: emulation ends

Statistics:

	average packet inter-arrival time time = 0.4133488
	average packet service time = 1.4400781

	average number of packets in Q1 = 3.5815673
	average number of packets in Q2 = 2.0191395e-05
	average number of packets in S1 = 0.83321571
	average number of packets in S2 = 0.73007427
	average time a packet spent in system = 4.7393993
	standard deviation for time spent in system = 1.7766845

	token drop probability = 0
	packet drop probability = 0

